distributed/protocol/cudf.py

```diff
@@ -8,14 +8,18 @@
 # all (de-)serializtion code lives in the cudf codebase
 # here we ammend the returned headers with `is_gpu` for
 # UCX buffer consumption
-@cuda_serialize.register((cudf.DataFrame, cudf.Series, cudf.groupby.groupby._Groupby))
+@cuda_serialize.register(
+    (cudf.DataFrame, cudf.Index, cudf.Series, cudf.groupby.groupby._Groupby)
```

Comment:
It seems that this file is no longer used after cudf-0.9 - `cudf.groupby.groupby._Groupby` will no longer work.

Reply:
Should we drop the file altogether then?

Reply:
> Should we drop the file altogether then?

I'm in support of that :)

Reply:
@quasiben @madsbk @pentschev, any thoughts on dropping this file or do we still need it?

Reply:
I'm fine with it, I don't know if it's still relevant but note that there's a comment about UCX:

```
# all (de-)serializtion code lives in the cudf codebase
# here we ammend the returned headers with `is_gpu` for
# UCX buffer consumption
```

Maybe @quasiben can chime in.

Reply:
Right - we should wait for Ben's opinion.  However, we do know [this file is most likely ignored](https://github.com/dask/distributed/blob/6a76ca7ac6db9f6243869057b6805f14c23e3811/distributed/protocol/__init__.py#L88) for recent cudf versions

Reply:
i think we can drop this file.  We kept it around when moving to 0.9.  The code for switching can be found here:
https://github.com/dask/distributed/blob/6a76ca7ac6db9f6243869057b6805f14c23e3811/distributed/protocol/__init__.py#L83-L91

So if we drop the file we should update the __init__

Reply:
Yeah I saw that comment to @pentschev. Though I'm not actually seeing any use of `is_gpu` in the code. Perhaps the comment is outdated?

Reply:
The comment is outdated.  `is_gpu` was used in the previous ucx-py version when you needed to explicitly receive into a GPU buffer https://github.com/dask/distributed/blob/master/distributed/comm/ucx.py#L85-L87

Reply:
> The comment is outdated. `is_gpu` was used in the previous ucx-py version when you needed to explicitly receive into a GPU buffer https://github.com/dask/distributed/blob/master/distributed/comm/ucx.py#L85-L87

I think `is_gpu` is still used to determine when to use `cuda_array()`:
https://github.com/dask/distributed/blob/master/distributed/comm/ucx.py#L139
https://github.com/dask/distributed/blob/master/distributed/comm/ucx.py#L169

Reply:
@madsbk doesn't that use the `__cuda_array_interface__` interface instead of `is_gpus` ?

Reply:
Sorry, I thought you talked about the UCX communication protocol where the writer sends ` is_gpu` to the reader.

But yes, cuDF doesn't need the `is_gpu` amendment anymore. 
However, I think we still need to register a  serialization function for `cudf.DataFrame, cudf.Index, cudf.Series` right?

Reply:
Accordingly to @rjzamora this is now handled internally within cuDF. So it sounds like we can drop the whole file.